### DISCUSSION TITLE
Is registerDeviceForRemoteMessages needed if auto-registration is not disabled?

### DISCUSSION THREAD

**Discussion by Awadi-R** [invertase/react-native-firebase#6605] [Q&A]
Hi,

We're seeing the following error for some users of our app, when calling `messaging().getToken()`:
```
You must be registered for remote messages before calling getToken, see messaging().registerDeviceForRemoteMessages().
```

However, we **do not** have a firebase.json file that disables auto-registration (using the `messaging_ios_auto_register_for_remote_messages` parameter). 

It's unclear to me whether all devices should call `registerDeviceForRemoteMessages()` on startup (as stated in the release notes [here](https://github.com/invertase/react-native-firebase/blob/8fb859fb000d5c2c6bbb81dffc685e94ef1ba350/docs/releases/v6.4.0.md#messaging-1)) or if that's only needed when auto-registration has been disabled (as implied by the console warning [here](https://github.com/invertase/react-native-firebase/blob/930abd6920f70c89bdc70a674ff93f080b3b968d/packages/messaging/lib/index.js#L284)).

I also wanted to verify, based on this [other issue](https://github.com/invertase/react-native-firebase/issues/4788#issuecomment-763564873) whether requesting permissions has anything to do with it. According to the [Apple Docs](https://developer.apple.com/documentation/uikit/uiapplication/1623078-registerforremotenotifications), it seems that prompting for permissions is only needed to show user-facing notifications, and we're so we're not calling `messaging().requestPermission()` because we're interested in non-user-facing data messages with FCM. But I wanted to check anyway if that might be to blame for our failed FCM auto-registration.

Thank you!

**Comment by mikehardy** [comment]
We get an large amount of issue traffic, so I'm hesitant to remove any of the messages that help developers get things working, even if it means a few not-quite-necessary messages for users with different setups.

In this case, I think it never hurts to just call the API, double-calling it does nothing wrong, and then your log is clear

**Comment by ice-cap0** [comment]
Edit: removed what was  a bug on our end. For some reason Push Notification capability was removed from Signing & Capabilities and we were getting the same error mentioned by op

**Reply by mikehardy** [reply]
Well, if there was a useful outcome, it was detection of that problem! That's pretty serious.
Note I think you need 14.12.0 if you are currently avoiding the use_frameworks stuff in v15 - https://github.com/invertase/react-native-firebase/commit/1c777417365db4ce127b2cbdb07ae1d2d56e2d95

**Reply by ice-cap0** [reply]
thx for the note @mikehardy you're everywhere! updating now
